api/apps/api/src/modules/geo-features/geo-features.service.ts

```diff
@@ -270,4 +280,91 @@ export class GeoFeaturesService extends AppBaseService<
   ): Promise<GeoFeature> {
     return entity;
   }
+
+  public async createFeaturesForShapefile(
+    projectId: string,
+    data: UploadShapefileDTO,
+    features: Record<string, any>[],
+  ): Promise<void> {
+    const [apiDbConnection, geoDbConnection] = Object.values(
+      DbConnections,
+    ).map((name) => getConnection(name));
```

Comment:
[minor] not sure if iterating gives us any profit there (could be just simple `const apiDbConnection = getConnection(DbConnections.Api...)`